src/pkg/resourceviews/resourcetypes.go

```diff
@@ -0,0 +1,91 @@
+package resourceviews
+
+import (
+	"fmt"
+
+	"github.com/brendank310/aztui/pkg/config"
+	"github.com/brendank310/aztui/pkg/layout"
+	"github.com/rivo/tview"
+)
+
+var resourceTypeSelectItemFuncMap = map[string]func(*ResourceTypeListView, string) tview.Primitive{
+	"SpawnAKSClusterListView":     (*ResourceTypeListView).SpawnAKSClusterListView,
+	"SpawnVirtualMachineListView": (*ResourceTypeListView).SpawnVirtualMachineListView,
+}
+
+func callResourceTypeMethodByName(view *ResourceTypeListView, methodName string, resourceType string) tview.Primitive {
+	// Check if the method exists in the map and call it with the receiver
+	if method, exists := resourceTypeSelectItemFuncMap[methodName]; exists {
+		return method(view, resourceType) // Call the method with the receiver
```

Comment:
My understanding is that you are actually calling that method (SpawnAKSClusterListView/SpawnVirtualMachineListView) here, but both are being called simultaneously in response to the enter key being pressed. There could be some check for a state where correct one is being called. Not sure if tview has a way to distinguish between the events on separate list items.

Reply:
I think the inputhandlers can be installed on a per widget (tview.Primitive) basis. I'll defer to Adrian on that.